### DISCUSSION TITLE
Significance of offline TTL

### DISCUSSION THREAD

**Discussion by maaz-yobe** [keygen-sh/community#143] [General]
I wanted to confirm how the TTL in an offline license/machine file work. If my license is lets say 2 months long and my offline file TTL is 1 month, does that mean my offline file will be unusable after a month? 
In this case, I'm comparing machine time against the start and end date of the license (2 months). In this case, the TTL should make no difference, correct?

**Comment by ezekg** [comment]
It encourages you to occasionally sync with the API. For example, the TTL can be used to enforce an offline time limit — the actual license may expire after 1 year, but if the license file expires after 2 weeks, then end-user must go online every 2 weeks to check out a new license file. Or if the license never expires, you could set the TTL to 1 year to enforce a yearly renewal schedule in accordance with your sales cycle.

This will allow you to validate the license every 2 weeks, to ensure it's still in compliance, e.g. revalidate it, check if it's suspended, sync new entitlements, etc. This is completely optional. If you don't need a TTL, you can set it to null.

**Reply by maaz-yobe** [reply]
Thank you this helps a lot!
